sufia-models/lib/sufia/models/stats/user_stat_importer.rb

```diff
@@ -38,8 +38,16 @@ def date_since_last_cache(user)
       end
     end
 
-    def files_for_user(user)
-      ::GenericFile.where(Solrizer.solr_name('depositor', :symbol) => user.user_key)
+    def file_ids_for_user(user)
+      page_size = 1000
+      ids = []
+      resp = ActiveFedora::SolrService.instance.conn.get "select", params:{fq:["#{Solrizer.solr_name("has_model", :symbol)}:\"info:fedora/afmodel:GenericFile\"", "#{Solrizer.solr_name('depositor', :symbol)}:\"#{user.user_key}\""],fl:["id", "#{Solrizer.solr_name('depositor', :symbol)}"], rows: page_size}
```

Comment:
Actually GenericFile.find_in_batches might be the best way, then you'd only need to pass in the depositor.